charms/jupyter-controller/src/charm.py

```diff
@@ -13,31 +13,42 @@
 log = logging.getLogger()
 
 
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
+
+    def __init__(self, msg, status=None):
+        super().__init__()
+
+        self.msg = msg
+        self.status = status
```

Comment:
Can you update this with the latest changes we added? You can see them [here](https://github.com/canonical/argo-operators/blob/2f7d7d4ce9cd3031e115a458a624f78cdd4d5aca/operators/argo-controller/src/charm.py#L23) and [here](https://github.com/canonical/argo-operators/blob/2f7d7d4ce9cd3031e115a458a624f78cdd4d5aca/operators/argo-controller/src/charm.py#L61)

Reply:
I believe it looks like the latest version of your PRs, let me know otherwise.